DuckDuckGo.Fluent.Plugin/DuckDuckGoSearchApp.cs

```diff
@@ -0,0 +1,195 @@
+        {
+            return _applicationInfo;
+        }
+
+        public async IAsyncEnumerable<ISearchResult> SearchAsync(SearchRequest searchRequest,
+            [EnumeratorCancellation] CancellationToken cancellationToken)
+        {
+            string searchedTag = searchRequest.SearchedTag;
+            string searchedText = searchRequest.SearchedText;
+
+            if (!VerifySearchedTerms(searchedText, searchedTag)) yield break;
+            searchedText = searchedText.Trim();
+
+            if (!string.IsNullOrWhiteSpace(searchedTag) && searchedTag.Equals(QrTag))
+            {
+                DuckDuckGoSearchResult duckGoSearchResult = await GetQrImage(searchedText);
+                if (duckGoSearchResult != null)
+                    yield return duckGoSearchResult;
+            }
+            else
+            {
+                string url = GetEndpointUrl(searchedText);
+                DuckDuckGoApiResult apiResult = await HttpCalls.GetApiResult(url);
+
+                if (apiResult == null) yield break;
+
+                var duckResultFactory = DuckResultFactory.Create(apiResult, searchedText);
+
+                // Get Answers
+                DuckResult answers = duckResultFactory.GetAnswers();
+                if (answers != null) yield return GetISearchResult(answers);
+
+                // Get Definitions if available
+                DuckResult dictionary = duckResultFactory.GetDefinition();
+                if (dictionary != null) yield return GetISearchResult(dictionary);
+
+                // Get Abstract in Text form
+                DuckResult abstractResult = duckResultFactory.GetAbstract();
+                if (abstractResult != null)
+                    yield return GetISearchResult(abstractResult);
+
+                // External Links associated with search like Official Website etc.
+                List<DuckResult> externalLinks = duckResultFactory.GetExternalLinks();
+                foreach (DuckResult link in externalLinks) yield return GetISearchResult(link);
+
+                // Internal Links associated with Search.
+                List<DuckResult> internalLinks = duckResultFactory.GetRelatedTopics();
+                foreach (DuckResult link in internalLinks) yield return GetISearchResult(link);
+            }
+        }
+
+        public ValueTask LoadSearchApplicationAsync()
+        {
+            var assembly = Assembly.GetExecutingAssembly();
+            const string resourceName = "DuckDuckGo.Fluent.Plugin.duck_logo.png";
+            var image = new Bitmap(assembly.GetManifestResourceStream(resourceName)!);
+            _logoImage = new BitmapImageResult(image);
+            return ValueTask.CompletedTask;
+        }
+
+        public async ValueTask<ISearchResult> GetSearchResultForId(object searchObjectId)
+        {
+            DuckResult duckResult = searchObjectId switch
+            {
+                string json => JsonSerializer.Deserialize<DuckResult>(json),
```

Comment:
Why would it be json?

Reply:
While debugging, I've found that for some unknown reasons, FS is only giving me the json in this `GetSearchResultForId` method rather than the `DuckResult` object.